exporter/signalfxexporter/dimensions/metadata.go

```diff
@@ -30,42 +30,56 @@ type MetadataUpdateClient interface {
 	PushMetadata([]*metadata.MetadataUpdate) error
 }
 
-var propNameSanitizer = strings.NewReplacer(
+var k8sLabelNameSanitizer = strings.NewReplacer(
 	".", "_",
 	"/", "_")
 
 func getDimensionUpdateFromMetadata(
 	metadata metadata.MetadataUpdate,
 	metricTranslator *translation.MetricTranslator,
 ) *DimensionUpdate {
-
-	translateDimension := func(dim string) string {
-		res := dim
-		if metricTranslator != nil {
-			res = metricTranslator.TranslateDimension(res)
-		}
-		return propNameSanitizer.Replace(res)
-	}
-
-	properties, tags := getPropertiesAndTags(metadata, translateDimension)
+	properties, tags := getPropertiesAndTags(metadata)
 
 	return &DimensionUpdate{
-		Name:       translateDimension(metadata.ResourceIDKey),
+		Name:       translateDimension(metricTranslator, metadata.ResourceIDKey),
 		Value:      string(metadata.ResourceID),
 		Properties: properties,
 		Tags:       tags,
 	}
 }
 
-func getPropertiesAndTags(
-	kmu metadata.MetadataUpdate,
-	translate func(string) string,
-) (map[string]*string, map[string]bool) {
+func translateDimension(metricTranslator *translation.MetricTranslator, dim string) string {
+	if metricTranslator != nil {
+		return metricTranslator.TranslateDimension(dim)
+	}
+	return dim
+}
+
+const (
+	oTelK8sLabelPrefix   = "k8s.label."
+	oTelK8sServicePrefix = "k8s.service."
+	sfxK8sServicePrefix  = "kubernetes_service_"
+)
+
+// sanitizeProperty processes any property key or tag to ensure conformity with SFx values.
+// Currently this method replaces "." and "/" in kubernetes labels and kubernetes service tag
+// with "_". This is done to ensure compatibility with properties sent by kubernetes-cluster
+// monitor in the SignalFx Agent.
+func sanitizeProperty(property string) string {
+	if strings.HasPrefix(property, oTelK8sLabelPrefix) {
+		return k8sLabelNameSanitizer.Replace(strings.Replace(property, oTelK8sLabelPrefix, "", 1))
+	} else if strings.HasPrefix(property, oTelK8sServicePrefix) {
+		return k8sLabelNameSanitizer.Replace(strings.Replace(property, oTelK8sServicePrefix, sfxK8sServicePrefix, 1))
+	}
+	return property
```

Comment:
Do we only want to run the sanitizer on properties/tags with these prefixes as we are doing here?  The comment describes sanitizing all of them.

Reply:
Those sanitization steps were only required for kubernetes labels collected as properties. However, after a conversation with @jrcamp we think it makes sense to send them as is since `.` and `/` are supported characters for properties.